connectors/byoc.py

```diff
@@ -650,6 +681,55 @@ async def document_count(self):
         )
         return result["count"]
 
+    async def check_configuration_formatting(self, fqn, service_type):
```

Comment:
I really like how you structured the code and it's super easy to read. However, the method's name is misleading because it not just checks but can update the configuration. 

**not a blocker**: I would also split the method into 2 because it does 2 things: validation, and modifying the doc.

Reply:
I broke it up into a wrapper function and two sub functions. Also added docstrings so everything was a bit clearer.